overlord/devicestate/firstboot_test.go

```diff
@@ -58,75 +59,82 @@ import (
 	"github.com/snapcore/snapd/timings"
 )
 
-type FirstBootTestSuite struct {
+type firstBootBaseTest struct {
 	testutil.BaseTest
 
 	systemctl *testutil.MockCmd
 
-	// TestingSeed helps populating seeds (it provides
-	// MakeAssertedSnap, WriteAssertions etc.) for tests.
-	*seedtest.TestingSeed
-
 	devAcct *asserts.Account
 
 	overlord *overlord.Overlord
 
 	perfTimings timings.Measurer
 }
 
-var _ = Suite(&FirstBootTestSuite{})
-
-func (s *FirstBootTestSuite) SetUpTest(c *C) {
-	s.BaseTest.SetUpTest(c)
+func (t *firstBootBaseTest) setupBaseTest(c *C, s *seedtest.SeedSnaps) {
+	t.BaseTest.SetUpTest(c)
 
 	tempdir := c.MkDir()
 	dirs.SetRootDir(tempdir)
-	s.AddCleanup(func() { dirs.SetRootDir("/") })
+	t.AddCleanup(func() { dirs.SetRootDir("/") })
 
-	s.AddCleanup(release.MockOnClassic(false))
+	t.AddCleanup(release.MockOnClassic(false))
 
 	// mock the world!
 	err := os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "snaps"), 0755)
 	c.Assert(err, IsNil)
-	err = os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "assertions"), 0755)
-	c.Assert(err, IsNil)
 
 	err = os.MkdirAll(dirs.SnapServicesDir, 0755)
 	c.Assert(err, IsNil)
 	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "1")
-	s.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
-	s.systemctl = testutil.MockCommand(c, "systemctl", "")
-	s.AddCleanup(s.systemctl.Restore)
-
-	err = ioutil.WriteFile(filepath.Join(dirs.SnapSeedDir, "seed.yaml"), nil, 0644)
-	c.Assert(err, IsNil)
+	t.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
+	t.systemctl = testutil.MockCommand(c, "systemctl", "")
+	t.AddCleanup(t.systemctl.Restore)
 
-	s.TestingSeed = &seedtest.TestingSeed{}
-	s.SetupAssertSigning("can0nical", s)
+	s.SetupAssertSigning("canonical")
 	s.Brands.Register("my-brand", brandPrivKey, map[string]interface{}{
 		"verification": "verified",
 	})
 
-	s.SnapsDir = filepath.Join(dirs.SnapSeedDir, "snaps")
-	s.AssertsDir = filepath.Join(dirs.SnapSeedDir, "assertions")
-
-	s.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
+	t.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
 		"account-id": "developerid",
 	}, "")
 
-	s.AddCleanup(ifacestate.MockSecurityBackends(nil))
+	t.AddCleanup(sysdb.InjectTrusted([]asserts.Assertion{s.StoreSigning.TrustedKey}))
+	t.AddCleanup(ifacestate.MockSecurityBackends(nil))
 
 	ovld, err := overlord.New(nil)
 	c.Assert(err, IsNil)
 	ovld.InterfaceManager().DisableUDevMonitor()
-	s.overlord = ovld
+	t.overlord = ovld
 	c.Assert(ovld.StartUp(), IsNil)
 
 	// don't actually try to talk to the store on snapstate.Ensure
 	// needs doing after the call to devicestate.Manager (which happens in overlord.New)
 	snapstate.CanAutoRefresh = nil
 
-	s.perfTimings = timings.New(nil)
+	t.perfTimings = timings.New(nil)
+}
+
+type firstBoot16Suite struct {
+	firstBootBaseTest
```

Comment:
after SU today, it's seems to me this refactoring is so that you can make it easier to share this definition with a (presumed) `firstBoot20Suite` but in my first start to the review it wasn't clear, maybe next time a comment in the description explaining "xyz bit will be used in the future with abc", etc. even if you don't have another PR open to point to exactly I think it would be helpful

Reply:
yes, there's a firstBoot20Suite is a follow up, and yes the description should have made that clear, sorry, a bit lost in my own world here

Reply:
I put that in the description now